plan/exhaust_physical_plans.go

```diff
@@ -266,7 +280,7 @@ func (p *LogicalJoin) getIndexJoinByOuterIdx(prop *requiredProp, outerIdx int) [
 		pkCol := x.getPKIsHandleCol()
```

Comment:
Can we remove the `len(innerJoinKeys) == 1` restriction, and handle the following situation (pk is handle):
```sql
drop table if exists t1, t2;
create table t1(a bigint primary key, b bigint);
create table t2(a bigint, b bigint);
select /*+ TIDB_INLJ(t2) */ * from t1 join t2 on t1.a=t2.a and t1.b=t2.b;
```

For now, we can use index join only when the `on` condition is `t1.a=t2.a`:
```sql
TiDB(localhost:4000) > desc select /*+ TIDB_INLJ(t2) */ * from t1 join t2 on t1.a=t2.a;
+----------------+-------------+------------------------------+------+----------------------------------------------------------------------------+----------+
| id             | parents     | children                     | task | operator info                                                              | count    |
+----------------+-------------+------------------------------+------+----------------------------------------------------------------------------+----------+
| TableScan_6    |             |                              | cop  | table:t1, range:[-inf,+inf], keep order:false                              | 10000.00 |
| TableReader_7  | IndexJoin_8 |                              | root | data:TableScan_6                                                           | 10000.00 |
| TableScan_9    |             |                              | cop  | table:t2, range:[-inf,+inf], keep order:false                              | 10000.00 |
| TableReader_10 | IndexJoin_8 |                              | root | data:TableScan_9                                                           | 10000.00 |
| IndexJoin_8    |             | TableReader_7,TableReader_10 | root | outer:TableReader_10, inner join, outer key:test.t2.a, inner key:test.t1.a | 12500.00 |
+----------------+-------------+------------------------------+------+----------------------------------------------------------------------------+----------+
5 rows in set (0.00 sec)
```

But for the `on` condition `t1.a=t2.a and t1.b=t2.b`, we still use the hash join:
```sql
TiDB(localhost:4000) > desc select /*+ TIDB_INLJ(t2) */ * from t1 join t2 on t1.a=t2.a and t1.b=t2.b;
+----------------+----------------+------------------------------+------+---------------------------------------------------------------------------------------------+----------+
| id             | parents        | children                     | task | operator info                                                                               | count    |
+----------------+----------------+------------------------------+------+---------------------------------------------------------------------------------------------+----------+
| TableScan_8    |                |                              | cop  | table:t1, range:[-inf,+inf], keep order:false                                               | 10000.00 |
| TableReader_9  | HashLeftJoin_6 |                              | root | data:TableScan_8                                                                            | 10000.00 |
| TableScan_10   |                |                              | cop  | table:t2, range:[-inf,+inf], keep order:false                                               | 10000.00 |
| TableReader_11 | HashLeftJoin_6 |                              | root | data:TableScan_10                                                                           | 10000.00 |
| HashLeftJoin_6 |                | TableReader_9,TableReader_11 | root | inner join, inner:TableReader_11, equal:[eq(test.t1.a, test.t2.a) eq(test.t1.b, test.t2.b)] | 12500.00 |
+----------------+----------------+------------------------------+------+---------------------------------------------------------------------------------------------+----------+
5 rows in set (0.00 sec)
```

Reply:
Oh, it seems to be easier than i think previously. I'll look into it.